spark/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerDeleter.scala

```diff
@@ -0,0 +1,52 @@
+package geotrellis.spark.io.accumulo
+
+import geotrellis.spark.io.index.KeyIndex
+import geotrellis.spark.{Boundable, KeyBounds, LayerId}
+import geotrellis.spark.io._
+import org.apache.accumulo.core.client.{BatchWriterConfig, Connector}
+import org.apache.accumulo.core.security.Authorizations
+import spray.json.JsonFormat
+import org.apache.accumulo.core.data.{Range => AccumuloRange}
+import org.apache.hadoop.io.Text
+import geotrellis.spark.io.json._
+import spray.json.DefaultJsonProtocol._
+import scala.collection.JavaConversions._
+import scala.reflect.ClassTag
+
+class AccumuloLayerDeleter[K: Boundable: JsonFormat: ClassTag]
+  (val attributeStore: AttributeStore[JsonFormat], connector: Connector) extends LayerDeleter[LayerId] {
+
+  def delete(id: LayerId): Unit = {
+    if (!attributeStore.layerExists(id)) throw new LayerNotFoundError(id)
+    val (header, _, keyBounds, keyIndex, _) = try {
+      attributeStore.readLayerAttributes[AccumuloLayerHeader, Unit, KeyBounds[K], KeyIndex[K], Unit](id)
+    } catch {
+      case e: AttributeNotFoundError => throw new LayerDeleteError(id).initCause(e)
+    }
+
+    def decompose(bounds: KeyBounds[K], keyIndex: KeyIndex[K]) =
+      keyIndex.indexRanges(bounds).map { case (min, max) =>
+        new AccumuloRange(new Text(long2Bytes(min)), new Text(long2Bytes(max)))
+      }
+
+    val ranges = decompose(keyBounds, keyIndex)
+    val numThreads = 1
+    val config = new BatchWriterConfig()
+    config.setMaxWriteThreads(numThreads)
+    val deleter = connector.createBatchDeleter(header.tileTable, new Authorizations(), numThreads, config)
+    deleter.fetchColumnFamily(columnFamily(id))
+    deleter.setRanges(ranges)
```

Comment:
I believe if you create a range from -inf to +inf will do the trick here:
https://github.com/geotrellis/geotrellis/pull/1224/files#diff-5c4305182b01c1f7c572a3e031799577R37